libp2p/muxers/yamux/yamux.nim

```diff
@@ -442,7 +441,7 @@ proc cleanupChannel(m: Yamux, channel: YamuxChannel) {.async: (raises: []).} =
   when defined(libp2p_yamux_metrics):
     libp2p_yamux_channels.set(
       m.lenBySrc(channel.isSrc).int64, [$channel.isSrc, $channel.peerId])
-  if channel.isReset and channel.recvWindow > 0:
+  if (channel.isReset or not channel.closedRemotely.completed()) and channel.recvWindow > 0:
```

Comment:
I'm not sure about this new condition.

Reply:
`channel.closedRemotely` is always completed after `channel.join()`.
In theory, the only possible case is if we try to cancel, but `cleanupChannel is async spawned, so we cannot cancel it.

Reply:
`channel.join()` is completed after `LPstream.closeImpl()` is called.
`LPStream.closeImpl()` is called by `actuallyClosed` [here](https://github.com/vacp2p/nim-libp2p/pull/1086/files#diff-fa0addf2f141e6d44cffe29109c7b4fa96b608c52425616af4ff1cf763976d8fR194-R197), and it's called after checking if `closedRemotely` is completed.

Reply:
True, reverting.